### PR TITLE
added user-specific secret messages

### PR DESCRIPTION
**Pull request by Footkick72**
messages come from the json located [here](https://gist.githubusercontent.com/Footkick72/7f440faf62f546684d63f303171302d7/raw/23629738f142876decffc638309b3b2c51bb6985/messages.json). This is obviously configurable.

### CODE REVIEW

File: misc-traits.js [hunk 1]
```diff
@@ -292,4 +292,66 @@ class OptimalComparisonTrait extends Trait {
...
+        if (this.stg(state.stats).uuid) {
+            let bin = this.stg(state.stats).uuid.toString(2);
+            console.log(this.stg(state.stats).uuid);
+            let invis = bin.replace(/0/g, '\u200B').replace(/1/g, '\u200D');
+            return invis;
+        }
+        return "";
+    }
+
+    onReload(state) {
```
Comment (cjbcoombes) [inline misc-traits.js:319]: Should call `super.onReload`

File: misc-traits.js [hunk 2]
```diff
@@ -292,4 +292,66 @@ class OptimalComparisonTrait extends Trait {
...
+    }
+
+    onSave(state) {
+        this.stg(state.stats).uuid = this.stg(state.data).uuid;
+    }
+    
+    onPreShare(state) {
+        if (this.stg(state.stats).uuid) {
+            let bin = this.stg(state.stats).uuid.toString(2);
+            console.log(this.stg(state.stats).uuid);
```
Comment (cjbcoombes) [inline misc-traits.js]: Remove console log in release

File: misc-traits.js [hunk 3]
```diff
@@ -292,4 +292,66 @@ class OptimalComparisonTrait extends Trait {
...
+            return invis;
+        }
+        return "";
+    }
+
+    onReload(state) {
+        const msg = this.getSecret(state);
+        msg.then(data => {
+            if (data) {
+                console.log(data);
```
Comment (cjbcoombes) [inline misc-traits.js]: Remove console log in release

File: retreiveUUID.py [hunk 4]
```diff
@@ -0,0 +1,3 @@
+raw = input() # copy the line right under achievements
```
Comment (cjbcoombes) [inline retreiveUUID.py]: Python, really? You couldn't have made this part in JavaScript too?

File: misc-traits.js [hunk 5]
```diff
@@ -292,4 +292,66 @@ class OptimalComparisonTrait extends Trait {
...
+                const box = docMake("div", ["overlay-box", "normal-text"]);
+                docMake("div", ["header-text", "gift-title"], box, e => {
+                    e.innerText = data;
+                    e.addEventListener("click", _e => {
```
Comment (cjbcoombes) [inline misc-traits.js:328]: Do we just let them figure out they have to click it to make it go away?

File: misc-traits.js [hunk 6]
```diff
@@ -292,4 +292,66 @@ class OptimalComparisonTrait extends Trait {
...
+        })
+    }
+
+    getSecret(state) {
+        const uuid = this.stg(state.stats).uuid;
+        const url = "https://gist.githubusercontent.com/Footkick72/7f440faf62f546684d63f303171302d7/raw/23629738f142876decffc638309b3b2c51bb6985/messages.json";
+        return fetch(url)
+            .then(response => {
+                if (!response.ok) {
+                    throw new Error('Network response was not ok: ' + response.statusText);
```
Comment (cjbcoombes) [inline misc-traits.js:343]: Probably fail more quietly

**Review by cjbcoombes — CHANGES_REQUESTED** [review]
I felt like adding a PR review since I haven't really done this stuff with GitHub before

File: misc-traits.js (hunk 5 above)
Comment (Footkick72) [reply misc-traits.js:328]: yes

Comment (Footkick72) [reply misc-traits.js:328]: it isn't that unintuitive

File: misc-traits.js (hunk 6 above)
Comment (Footkick72) [reply misc-traits.js:343]: it does in fact fail quietly

Comment (cjbcoombes) [reply misc-traits.js:343]: Still errors to the console, but fine whatever

File: retreiveUUID.py (hunk 4 above)
Comment (cjbcoombes) [reply retreiveUUID.py]: No Python

Comment (Footkick72) [reply retreiveUUID.py]: ok no python for you then

**Review by cjbcoombes — APPROVED** [review]

**Pull request merged**